test/server_xml_search.cpp

```diff
@@ -0,0 +1,957 @@
+
+#define CPPHTTPLIB_ZLIB_SUPPORT 1
+#include "./httplib.h"
+#include "gtest/gtest.h"
+
+#define SERVER_PORT 8201
+#include "server_testing_tools.h"
+
+std::string makeSearchResultsXml(const std::string& header,
+                                 const std::string& results)
+{
+  const char SEARCHRESULTS_XML_TEMPLATE[] = R"XML(<?xml version="1.0" encoding="UTF-8"?>
+<rss version="2.0"
+     xmlns:opensearch="http://a9.com/-/spec/opensearch/1.1/"
+     xmlns:atom="http://www.w3.org/2005/Atom">
+  <channel>
+    %HEADER%%RESULTS%
+  </channel>
+</rss>
+)XML";
+
+  std::string html = removeEOLWhitespaceMarkers(SEARCHRESULTS_XML_TEMPLATE);
+  html = replace(html, "%HEADER%", header);
+  html = replace(html, "%RESULTS%", results);
+  return html;
+}
+
+#define SEARCH_RESULT(LINK, TITLE, SNIPPET, BOOK_TITLE, WORDCOUNT) \
+"      <title>" TITLE "</title>\n"\
+"      <link>" LINK "</link>\n"\
+"        <description>" SNIPPET "</description>\n"\
+"        <book>\n"\
+"          <title>" BOOK_TITLE "</title>\n"\
+"        </book>\n"\
+"        <wordCount>" WORDCOUNT "</wordCount>"
+
+
+const std::vector<std::string> LARGE_SEARCH_RESULTS = {
+  SEARCH_RESULT(
+    /*link*/       "/ROOT/zimfile/A/Genius_+_Soul_=_Jazz",
+    /*title*/      "Genius + Soul = Jazz",
+    /*snippet*/    R"SNIPPET(...Grammy Hall of Fame in 2011. It was re-issued in the UK, first in 1989 on the Castle Communications "Essential Records" label, and by Rhino Records in 1997 on a single CD together with Charles' 1970 My Kind of <b>Jazz</b>. In 2010, Concord Records released a deluxe edition comprising digitally remastered versions of Genius + Soul = <b>Jazz</b>, My Kind of <b>Jazz</b>, <b>Jazz</b> Number II, and My Kind of <b>Jazz</b> Part 3. Professional ratings Review scores Source Rating Allmusic link Warr.org link Encyclopedia of Popular Music...)SNIPPET",
+    /*bookTitle*/  "Ray Charles",
+    /*wordCount*/  "242"
+  ),
+
```

Comment:
We better avoid this huge duplication between server_xml_search.cpp and server_html_search.cpp test suites

Reply:
I totally agree with you. But I think it is better for now to merge the PR as it is now to avoid too long review process (as long as the code is working).
But we need to do that in next future.